server/src/main/scala/org/http4s/server/package.scala

```diff
@@ -3,12 +3,47 @@ package org.http4s
 import scalaz.concurrent.Task
 
 package object server {
-  type Middleware  = HttpService => HttpService
-  type HttpService = Service[Request, Response]
+  type Service[-A, +B] = A => Task[B]
+
+  type PartialService[-A, +B] = Service[A, Option[B]]
 
+  type HttpService = Service[Request, Response]
   object HttpService {
-    def apply(pf: PartialFunction[Request, Task[Response]]): HttpService = Service(pf)
+    /**
+     * Lifts a partial function to a service.  The resulting service will
+     * return a Task containing `None` where the partial function is not
+     * defined, and Some result where it is.
+     */
+    def apply(pf: PartialFunction[Request, Task[Response]]): HttpService =
+      pf.lift.andThen {
+        case Some(respTask) => respTask
+        case None => Task.now(Response(Status.NotFound))
+      }
+
+    val empty: HttpService = Function.const(Task.now(Response(Status.NotFound)))
   }
+
+  implicit class HttpServiceSyntax(val service: HttpService) extends AnyVal {
```

Comment:
I think it must be a syntax unless `HttpService` goes back to a wrapper instead of a type alias.  And even then, we'd either lose `Service` or need evidence parameters for an `orElse` operation.